src/standalone/Vegetation/pmodel.jl

```diff
@@ -0,0 +1,317 @@
+export PModelParameters,
+    PModelDrivers, PModelConstants, compute_full_pmodel_outputs, PModel
+
+"""
+    PModelParameters{FT<:AbstractFloat}
+
+The required parameters for P-model (Stocker et al. 2020). Parameters are typically
+tunable with considerable uncertainty. 
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct PModelParameters{FT <: AbstractFloat}
+    "Constant describing cost of maintaining electron transport (unitless)
+    Typical value = 0.41"
+    cstar::FT
+    "Ratio of unit costs of transpiration and carboxylation (unitless)
+    Typical value = 146"
+    β::FT
+    "Scaling parameter for temp-dependent intrinsic quantum yield (unitless)
+    Typical value = 0.087"
+    ϕc::FT
+    "Temp-independent intrinsic quantum yield. If provided, overrides ϕc. (unitless)
+    Typical value = 0.05"
+    ϕ0::FT
+    """Constant term in temp-dependent intrinsic quantum yield (unitless)."""
+    ϕa0::FT
+    """First order term in temp-dependent intrinsic quantum yield (K^-1)."""
+    ϕa1::FT
+    """Second order term in temp-dependent intrinsic quantum yield (K^-2)."""
+    ϕa2::FT
+end
+
+"""
+    PModelDrivers{FT<:AbstractFloat}
+
+The required drivers for P-model (Stocker et al. 2020). Drivers are defined as
+external variables used to compute the optimal photosynthetic capacities. 
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct PModelDrivers{FT <: AbstractFloat}
```

Comment:
My [previous comment](https://github.com/CliMA/ClimaLand.jl/pull/1198#discussion_r2178002371) still stands - we can access these fields from Y and p instead of copying them to another location and accessing them from here. Instead of calling `compute_pmodel_outputs(parameters, drivers, constants)`, it could be something like `compute_pmodel_outputs(Y, p, parameters, constants)`